From c50254799a47ee436a85cdd945c4b0376ba545da Mon Sep 17 00:00:00 2001
From: Anoop C S <anoopcs@samba.org>
Date: Tue, 22 Sep 2026 16:26:26 +0530
Subject: [PATCH] s3/smbconf: Avoid leaking a stack frame for a NULL share name

Allocate the stack frame after the NULL servicename check so the
early return in smbconf_reg_create_share() does not leak it.

BUG: https://bugzilla.samba.org/show_bug.cgi?id=16258

Signed-off-by: Anoop C S <anoopcs@samba.org>
Reviewed-by: Volker Lendecke <vl@samba.org>

Autobuild-User(master): Volker Lendecke <vl@samba.org>
Autobuild-Date(master): Wed Sep 30 21:26:41 UTC 2026 on atb-devel-224

(cherry picked from commit c670e4fce53c04d3d41a47af403e771f34250fe1)
---
 source3/lib/smbconf/smbconf_reg.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/source3/lib/smbconf/smbconf_reg.c b/source3/lib/smbconf/smbconf_reg.c
index fc8aa62d0678..c1bc762dfe58 100644
--- a/source3/lib/smbconf/smbconf_reg.c
+++ b/source3/lib/smbconf/smbconf_reg.c
@@ -861,12 +861,14 @@ static sbcErr smbconf_reg_create_share(struct smbconf_ctx *ctx,
 {
 	sbcErr err;
 	struct registry_key *key = NULL;
-	TALLOC_CTX *tmp_ctx = talloc_stackframe();
+	TALLOC_CTX *tmp_ctx = NULL;
 
 	if (servicename == NULL) {
 		return SBC_ERR_OK;
 	}
 
+	tmp_ctx = talloc_stackframe();
+
 	err = smbconf_reg_create_service_key(tmp_ctx, ctx,
 					     servicename, &key);
 
